internal/controller/promotions/promotions.go

```diff
@@ -506,13 +507,14 @@ func (r *reconciler) promote(
 	}
 
 	promoCtx := directives.PromotionContext{
-		WorkDir:         filepath.Join(os.TempDir(), "promotion-"+string(workingPromo.UID)),
-		Project:         stageNamespace,
-		Stage:           stageName,
-		FreightRequests: stage.Spec.RequestedFreight,
-		Freight:         *workingPromo.Status.FreightCollection.DeepCopy(),
-		StartFromStep:   promo.Status.CurrentStep,
-		State:           directives.State(workingPromo.Status.GetState()),
+		APIServerBaseURL: r.cfg.APIServerBaseURL,
```

Comment:
Nit (and non-blocking): this should probably become part of a nested something at some point. As its placement feels odd as-is.

Reply:
Second nit: from the perspective of what we use it for now (within the context of directives), isn't this just the `UIBaseURL`?

Reply:
Fair point. I'll change that.

Reply:
Fixed in fd78be32.